### ISSUE TITLE
Don't add user to a domain invitation if we don't have a connection to SES (mail infra)

### ISSUE THREAD

**Issue by vickyszuchin** [cisagov/getgov#1780]
### Issue description

If there's an issue with our connection to SES, we shouldn't add a user to a domain invitation. However, right now users get added. (See screenshot on #1779)

[Slack thread ](https://cisa-corp.slack.com/archives/C05BGB4L5NF/p1707929221903979)of discussion

### Acceptance criteria

- [ ] Add a user to the domain invitation table only after confirming the connection to SES is successful

### Additional context

This issue doesn't attempt to solve the situation where a user types in an email address to a non-existent mailbox or some other SMTP failure (which was part of the original issue). 

Right now we add a new DomainInvitation object before trying to send the email (note the try except block in the code). This ticket is just to move the DomainInvitation creation to after we try to send it. You could use an "else" block in the try/except or you could just move the domain invitation below the code line that actually sends the email and is within the try block.

### Links to other issues

Related: #1779

**Comment by abroddrick** [comment]
refinement question: Should we change the domain invitation status to Failed to Send or something? That way if an analyst is dealing with a help desk question about stuff not sending they would be able to look and see clearly, "yep it hit an error" or "no it should have been sending, maybe the email is wrong". I don't think this is a very common thing since we are using aws email sending which is pretty reliable. CC: @Katherine-Osos

**Comment by h-m-f-t** [comment]
Making new states sounds about right, @abroddrick. It does seem like the likelier failure mode is sending to an bad email address that throws an SMTP 550 or something. Can we get SMTP errors from SES?

**Comment by vickyszuchin** [comment]
Per request from PO, added to refinement fireside meeting on 3/14.

**Comment by h-m-f-t** [comment]
In product huddle, we determined that this issue should be revised to focus just on the SES infrastructure connection. Note the comment in additional context, which might be future work.
